doc/drop_counters/drop_counters_test_plan.md

```diff
@@ -0,0 +1,584 @@
+
+We should also send some packets to that interface before shutting it down to ensure that everything is forwarding correctly before we modify the system.
+
+## 4.3 Multiple Pipeline Drops
+
+### 4.3.1. Verify that a counter with multiple drop reasons still works correctly
+#### Counters:
+```
+TEST_COUNTER
+    Type: INGRESS
+    Reasons: SMAC_EQUALS_DMAC, SIP_LINK_LOCAL, DIP_LINK_LOCAL
+```
+
+#### Sample Packets:
+Packet A:
+```
+###[ Ethernet ]###
+  dst = [DUT_MAC_ADDR]
+  src = [DUT_MAC_ADDR]
+  type = 0x800
+...
+```
+
+Packet B:
+```
+###[ Ethernet ]###
+  dst = [auto]
+  src = [auto]
+  type = 0x800
+###[ IP ]###
+    version = 4
+    ttl = [auto]
+    proto = tcp
+    src = 169.254.0.0
+    dst = 2.2.2.2
+...
+```
+
+#### Additional Steps: N/A
+
+### 4.3.2. Verify that a packet that fulfills multiple drop reasons is only counted once
+#### Counters:
+```
+TEST_COUNTER
+    Type: INGRESS
+    Reasons: SMAC_EQUALS_DMAC, SIP_LINK_LOCAL, DIP_LINK_LOCAL
+```
+
+#### Sample Packets:
+```
+###[ Ethernet ]###
+  dst = [DUT_MAC_ADDR]
+  src = [DUT_MAC_ADDR]
+  type = 0x800
+###[ IP ]###
+    version = 4
+    ttl = [auto]
+    proto = tcp
+    src = 169.254.0.0
+    dst = 2.2.2.2
+...
+```
+
+#### Additional Steps: N/A
+
+## 4.4 Multiple Counters
+
+### 4.4.1. Verify that two drop counters with different drop reasons both count drops correctly
+#### Counters:
+```
+TEST_COUNTER_0
+    Type: INGRESS
+    Reasons: SMAC_EQUALS_DMAC
+
+TEST_COUNTER_1
+    Type: INGRESS
+    Reasons: SIP_LINK_LOCAL, DIP_LINK_LOCAL
+```
+
+#### Sample Packets:
+Packet A:
+```
+###[ Ethernet ]###
+  dst = [DUT_MAC_ADDR]
+  src = [DUT_MAC_ADDR]
+  type = 0x800
+...
+```
+
+Packet B:
+```
+###[ Ethernet ]###
+  dst = [auto]
+  src = [auto]
+  type = 0x800
+###[ IP ]###
+    version = 4
+    ttl = [auto]
+    proto = tcp
+    src = 169.254.0.0
+    dst = 2.2.2.2
+...
+```
+
+#### Additional Steps:
+Because the counters are non-overlapping, the sum of the two counters should yield the total number of drops when we perform the calculation.
+
+### 4.4.2. Verify that two drop counters with overlapping drop reasons both count drops correctly
+#### Counters:
+```
+TEST_COUNTER_0
+    Type: INGRESS
+    Reasons: SMAC_EQUALS_DMAC
+
+TEST_COUNTER_1
+    Type: INGRESS
+    Reasons: SMAC_EQUALS_DMAC, SIP_LINK_LOCAL, DIP_LINK_LOCAL
+```
+
+#### Sample Packets:
+Packet A:
+```
+###[ Ethernet ]###
+  dst = [DUT_MAC_ADDR]
+  src = [DUT_MAC_ADDR]
+  type = 0x800
+...
+```
+
+Packet B:
+```
+###[ Ethernet ]###
+  dst = [auto]
+  src = [auto]
+  type = 0x800
+###[ IP ]###
+    version = 4
+    ttl = [auto]
+    proto = tcp
+    src = 169.254.0.0
+    dst = 2.2.2.2
+...
+```
+
+#### Additional Steps:
+Because counter B includes all the drop reasons, the total in counter B should be equal to the total number of drops. Counter A should only be equal to the number of SMAC_EQUALS_DMAC packets sent.
```

Comment:
Can you please clarify a little bit more about "total number of drops" in 
"the total in counter B should be equal to the total number of drops."

Reply:
Updated.